src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -67,20 +83,20 @@ def create_launcher(fluent_launch_mode: LaunchMode = None, **kwargs):
         If an unknown Fluent launch mode is passed.
     """
     allowed_options = [mode for mode in LaunchMode]
-    if (
-        not isinstance(fluent_launch_mode, LaunchMode)
-        or fluent_launch_mode not in allowed_options
-    ):
+    if fluent_launch_mode not in allowed_options:
         raise DisallowedValuesError(
             "fluent_launch_mode",
             fluent_launch_mode,
             allowed_values=allowed_options,
         )
     if fluent_launch_mode == LaunchMode.STANDALONE:
+        kwargs = _remove_unused_args(fluent_launch_mode, **kwargs)
```

Comment:
Any invalid arguments can raise a `TypeError` automatically.

Reply:
It raises `__init__ got an unexpected keyword argument` therefore it is necessary to filter unused arguments before passing to the constructor.

Reply:
We have removed this code now. Thank you.